teps/0129-multiple-tekton-instances-per-cluster.md

```diff
@@ -0,0 +1,312 @@
+---
+status: proposed
+title: Multiple Tekton instances per cluster
+creation-date: '2023-01-11'
+last-updated: '2023-01-11'
+authors:
+- '@afrittoli'
+collaborators: []
+---
+
+# TEP-0129: Multiple Tekton instances per cluster
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+  - [Use Cases](#use-cases)
+  - [Requirements](#requirements)
+- [Proposal](#proposal)
+  - [Notes and Caveats](#notes-and-caveats)
+- [Design Details](#design-details)
+- [Design Evaluation](#design-evaluation)
+  - [Reusability](#reusability)
+  - [Simplicity](#simplicity)
+  - [Flexibility](#flexibility)
+  - [User Experience](#user-experience)
+  - [Performance](#performance)
+  - [Risks and Mitigations](#risks-and-mitigations)
+  - [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+- [Implementation Plan](#implementation-plan)
+  - [Test Plan](#test-plan)
+  - [Infrastructure Needed](#infrastructure-needed)
+  - [Upgrade and Migration Strategy](#upgrade-and-migration-strategy)
+  - [Implementation Pull Requests](#implementation-pull-requests)
+- [References](#references)
+<!-- /toc -->
+
+## Summary
+
+With the growth of [Tekton adoption][tekton-adoption], Tekton is now used as
+a foundation technology for several open source tools as well as commercial
+offerings.
+
+Users may want to adopt different tools that rely on Tekton and run them in
+the same kubernetes clusters. Since tools may not be aligned in the version
+and configuration of Tekton they depend on, the main roadblock today is the
+ability to run multiple different instances of Tekton in a single kubernetes
+cluster.
+
+## Motivation
+
+The Tekton controller provides some functionality towards the ability of
+running multiple instances in a single cluster, namely the ability to
+configure the Tekton controller to watch a single namespace for resources
+to be reconciled. This functionality is not properly documented and it does
+not include other Tekton controllers, such as the various controllers
+bundled in the Tekton webhook as well the remote resolution controller.
+
+### Goals
+
+- Allow multiple instances of Tekton Pipeline to run in single kubernetes
+  cluster, each responsible for a Tekton resources in a dedicated namespaces. Each
+  instance performs validation, resolution and execution and is configured
+  independently from the others. 
+- Allow multiple version of Tekton Pipeline to run in a single kubernetes cluster,
+  as long as there are no backward incompatible changes between the older and newer
+  versions.
+- Identify and configure cluster level resources in Tekton Pipelines, such as CRDs,
+  cluster roles and bindings and the conversion webhook, so that they can safely
+  shared across multiple Tekton instances.
+
+### Non-Goals
+
+- Support multiple instances of Tekton Triggers
+- Changes to the Tekton cli, dashboard, operator, results, chains, triggers to
```

Comment:
We may want to include the operator here, to make sure that uninstalling an operator-installed Tekton does not break other instances of Tekton running in the same cluster. 
Alternatively we could exclude operator-installed tekton in the first iteration and add operator support as a follow-up